Automattic/wp-calypso
# Add Google one tap prompt

This adds Google One Tap prompt login hidden behind a flag. Context is here: peP6yB-29m-p2
<img width="1051" alt="Screenshot 2024-09-24 at 17 26 24" src="https://github.com/user-attachments/assets/369bd119-ba0f-4dd3-acb9-d7f3ddb10fb1">

Related to #

## Proposed Changes


## Why are these changes being made?


* To make signing in with Google smoother.



## Testing Instructions



This should be tested locally.

1. Checkout this branch.
2. Run `yarn start-mock-wordpress-com`.
3. Add `127.0.0.1 wordpress.com` to your hosts file.
4. Sandbox public-api.wordpress.com.
5. Sandbox de.wordpress.com.

### Testing login
1. Open incognito window
2. Visit https://wordpress.com/log-in?flags=google_one_tap
3. The login page should look the same
4. Login to a google account in another tab
5. Reload the login page. You should see the prompt.
6. By clicking continue on the prompt you should be logged in successfully.

### Testing user social step
1. Close all incognito windows and open a new incognito window
2. Visit https://wordpress.com/start/user-social?flags=google_one_tap
3. The page should look the same
4. Login to a google account in another tab
5. Reload the user step page. You should see the prompt.
6. By clicking continue on the prompt you should be logged in successfully.


## Pre-merge Checklist
